yt/data_objects/tests/test_particle_trajectories.py

```diff
@@ -21,28 +16,36 @@ def setup():
 pfields = ["particle_position_x", "particle_position_y", "particle_position_z"]
 vfields = ["particle_velocity_x", "particle_velocity_y", "particle_velocity_z"]
 
-@requires_ds("Orbit/orbit_hdf5_chk_0000")
 def test_orbit_traj():
-    fields = ["particle_velocity_x", "particle_velocity_y", "particle_velocity_z"]
-    my_fns = glob.glob(os.path.join(data_path, "Orbit/orbit_hdf5_chk_00[0-9][0-9]"))
-    my_fns.sort()
-    ts = DatasetSeries(my_fns)
+    n_particles = 16 ** 2
+    ids = np.arange(n_particles, dtype=int) % (n_particles // 2)
+    data = {'particle_index': ids}
+    fields = ['particle_type', 'density', 'particle_position_x',
+              'particle_position_y', 'particle_position_z', 'particle_index']
+    units = ['dimensionless', 'g/cm**3', 'cm', 'cm', 'cm', '1']
+    ts = DatasetSeries([fake_particle_ds(fields=fields, negative=[False] * 6,
+                                         units=units, npart=n_particles,
+                                         data=data) for i in range(5)])
     ds = ts[0]
     traj = ts.particle_trajectories([1, 2], fields=fields, suppress_logging=True)
     for field in pfields+vfields:
         def field_func(name):
             return traj[field]
         yield GenericArrayTest(ds, field_func, args=[field])
```

Comment:
Since it's the time series just consists of the same dataset over and over again, the trajectories should be trivial (e.g. all the particles are stationary). We could probably avoid the need to use the answer testing framework here and just compare with the known result that the particles should all be stationary.

Reply:
I do agree with the remark. Even though the lines 
```python
ts = DatasetSeries([fake_particle_ds(fields=fields, negative=[False] * 6,
                                     units=units, npart=n_particles,
                                     data=data) for i in range(5)])
```
will produce the same dataset over and over, it is far from obvious reading the code. If you want to have the same dataset, you should fill the `data` argument with particle positions.

One suggestion here would be something like
```python
import yt
from yt.testing import fake_particle_ds
from yt.data_objects.time_series import DatasetSeries
import numpy as np

N_particles = 16**2
N_ds = 5
ids = np.arange(N_particles)
pos = np.random.rand(N_particles) / 2

data = [
    {'particle_index': ids.copy(),
     'particle_position_x': pos.copy() * 2,
     'particle_position_y': pos.copy() + 0.5*i/N_ds,
     'particle_position_z': pos.copy() + 0.5*(1-i/N_ds)}
    for i in range(N_ds)
]

fields = ['particle_position_x', 'particle_position_y', 'particle_position_z', 'particle_index', 'particle_mas']
field_units = ['cm', 'cm', 'cm', '1', 'g']
ts = DatasetSeries([
    fake_particle_ds(fields=fields, units=field_units, data=dt,
                     negative=[False]*len(fields),
                     npart=N_particles)
    for dt in data])

traj = ts.particle_trajectories(ids)

assert np.allclose(np.diff(traj['particle_position_x'], axis=1), 0)
assert np.allclose(np.diff(traj['particle_position_y'], axis=1), 0.1)
assert np.allclose(np.diff(traj['particle_position_z'], axis=1), -0.1)
```
Note that in the process, I noticed that https://github.com/yt-project/yt/blob/master/yt/testing.py#L288 is actually preventing the `fake_particle_dataset` from being fed by arbitrary data.